WordPress/Classes/ViewRelated/Site Creation/FinalAssembly/LandInTheEditorHelper.swift

```diff
@@ -0,0 +1,39 @@
+typealias HomepageEditorCompletion = () -> Void
+
+class LandInTheEditorHelper {
+    /// Land in the editor, or continue as usual for the control group -  Used to branch on the ExPlat experiment for landing in the editor from the site creation flow
+    /// - Parameter blog: Blog (which was just created) for which to show the home page editor
+    /// - Parameter navigationController: UINavigationController used to present the home page editor
+    /// - Parameter completion: HomepageEditorCompletion callback to be invoked after the user finishes editing the home page, or immediately in the control group case
+    static func landInTheEditorOrContinue(for blog: Blog, navigationController: UINavigationController, completion: @escaping HomepageEditorCompletion) {
+        // branch here for explat variation
+        if ABTest.landInTheEditorPhase1.variation == .control {
+            completion()
+        } else {
+            landInTheEditor(for: blog, navigationController: navigationController, completion: completion)
+        }
+    }
+
+    private static func landInTheEditor(for blog: Blog, navigationController: UINavigationController, completion: @escaping HomepageEditorCompletion) {
+        fetchAllPages(for: blog, success: { _ in
+            DispatchQueue.main.async {
+                if let homepage = blog.homepage {
+                    let editorViewController = EditPageViewController(homepage: homepage, completion: completion)
+                    navigationController.present(editorViewController, animated: false)
+                    WPAnalytics.track(.landingEditorShown)
+                }
+            }
+        }, failure: { _ in
+            NSLog("Fetching all pages failed after site creation!")
+        })
+    }
+
+    // This seems to be necessary before casting `AbstractPost` to `Page`.
+    private static func fetchAllPages(for blog: Blog, success: @escaping PostServiceSyncSuccess, failure: @escaping PostServiceSyncFailure) {
+        let options = PostServiceSyncOptions()
+        options.number = 20
```

Comment:
I wonder if the fetch limit is needed here (I see that it is [also used in pages list](https://github.com/wordpress-mobile/WordPress-iOS/blob/a08cc72e4bfb15d4fa88c06c549c16941fb2a64a/WordPress/Classes/ViewRelated/Pages/PageListViewController.swift#L265)) but I think the Android side does not use it.

Reply:
If left out, I think the default for .com REST API is 20 anyway, and for WP API I believe it is 10, so this makes them the same. Otoh, it shouldn't make a difference until / unless one of the starter sites had more than 10 pages (and for the usage here, the additional condition that the home page was the 11th page or after.

Reply:
Thank you for the extra information @mkevins 🙇 
> I think it's safest to leave this in.
I agree with that 👍